getnacelle/nacelle-js
# feat: upgrade to latest Shopify Storefront API

## Why are these changes introduced?

Addresses [ENG-8375](https://nacelle.atlassian.net/browse/ENG-8375) 



> Upgrade Shopify Cart package to use 2023-01 Shopify Storefront API

## What is this pull request doing?



This PR upgrades `@nacelle/shopify-cart`'s Shopify Storefront API version from `2022-10` to `2023-01`, which is now the latest stable API version.

## How to Test



1. Check out the `ENG-8375-shopify-storefront-api-version-2023-01` branch.
2. Navigate to `packages/shopify-cart`.
3. `npm run build` - there should be no errors.
4. `npm run test:ci` - all tests should pass with 100% coverage.
5. `npm run codegen` - there should be no uncommitted changes.

In local testing, I found that the Next.js Reference Store made requests with the `2023-01` API version, and all cart CRUD actions worked as expected:

https://user-images.githubusercontent.com/5732000/212422927-c5f84d5e-9b27-447f-bf70-75e79cfeff86.mov

## Checklist



[ENG-8375]: https://nacelle.atlassian.net/browse/ENG-8375?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
